src/xtgeo/grid3d/grid.py

```diff
@@ -567,15 +591,14 @@ def ijk_handedness(self):
         return self._ijk_handedness
 
     @ijk_handedness.setter
-    def ijk_handedness(self, value):
-        if value in ("right", "left"):
-            self.reverse_row_axis(ijk_handedness=value)
-        else:
+    def ijk_handedness(self, value: Literal["left", "right"]) -> None:
+        if value not in ("right", "left"):
             raise ValueError("The value must be 'right' or 'left'")
+        self.reverse_row_axis(ijk_handedness=value)
         self._ijk_handedness = value
 
     @property
-    def subgrids(self):
+    def subgrids(self) -> Optional[OrderedDict[str, range | list[int]]]:
```

Comment:
I think this type is still not fully correct, it should be

`Optional[OrderedDict[str, range], OrderedDict[str, list[int]]]` , two possible dict types not a single one with two possible values for keys. Relevant for the deprecated init, init, setter, maybe one or two others places. Then you could get rid of the `# type: ignore` in `subgrids_from_zoneprop`